cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -469,7 +540,16 @@ fn insert_common_prologue(
         pos.func.locations[csr_arg] = ir::ValueLoc::Reg(reg);
 
         // Remember it so we can push it momentarily
-        pos.ins().x86_push(csr_arg);
+        let reg_push_inst = pos.ins().x86_push(csr_arg);
+        cfa_state.position -= word_size;
+        pos.func.frame_layout.instructions.insert(
+            reg_push_inst,
+            vec![FrameLayoutChange::RegAt {
+                reg: reg,
```

Comment:
Are we interested in the position of every register? (I guess so, if I understand correctly this is for tracking where local variables are.) If so, should every instruction in the function's body also have its own frame layout changes (in case regalloc moves a variable from one register to another)?

Reply:
With this PR (and .debug_frame) we are interested in basics: proper frame unwinding (i.e. BP, SP, IP) and maybe registers used for args. (The tracking of original variables values is happening somewhere else).